### ISSUE TITLE
Modulo (%) operator not supported anymore for Index since 0.15

### ISSUE THREAD

**Issue by jorisvandenbossche** [pydata/pandas#9244]
In 0.14:

```
In [1]: pd.__version__
Out[1]: '0.14.1'

In [2]: intidx = pd.Index(range(10))

In [3]: intidx % 2
Out[3]: Int64Index([0, 1, 0, 1, 0, 1, 0, 1, 0, 1], dtype='int64')
```

now:
```
In [1]: pd.__version__
Out[1]: '0.15.2'

In [2]: intidx = pd.Index(range(10))

In [3]: intidx % 2
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-de0ac1282c31> in <module>()
----> 1 intidx % 2

TypeError: unsupported operand type(s) for %: 'Int64Index' and 'int'
```

**Comment by jreback** [comment]
changeover from ``ndarray`` did it. 

really what should happen is ``NumericIndex`` get these special methods thru ``core/ops.py`` (like Series does now).

**Comment by nbonnotte** [comment]
@jreback I've been looking into this, and into your suggestion, but the thing is that `core/ops.py` imports `pandas.index`, so `NumericIndex` cannot directly get these specials methods from `ops`.

I'll start with a simple PR with just the definition of `NumericIndex.__mod__`. That should do it.

**Comment by jreback** [comment]
this is quite straightforward, just update ``pandas/core/index.py/Index/_add_numeric_methods``

the imports are not an issue

**Comment by nbonnotte** [comment]
Straightforward indeed, thx!

**Comment by jreback** [comment]
closed by #12012
